Api/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,175 @@
+                                  ILogger<CategoryController> logger,
+                                  LinkGenerator linkGenerator,
+                                  IProductService productService)
+        {
+            _categoryService = categoryService;
+            _logger = logger;
+            _linkGenerator = linkGenerator;
+            _productService = productService;
+        }
+
+
+
+        // GET: api/<CategoryController>
+        [HttpGet]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(LinkCollectionWrapper<CategoryModel>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> Get()
+        {
+            var categoriesDto = await _categoryService.GetAllCategoriesAsync();
+            var categoryModels = categoriesDto.Select(CategoryModel.MapFrom).ToList();
+
+            foreach (var category in categoryModels)
+            {
+                var links = CreateLinksForCategory(category.Id);
+                category.Links = links.ToList();
+            }
+
+            var categoriesWrapper = new LinkCollectionWrapper<CategoryModel>(categoryModels);
+            return Ok(CreateLinksForGetAllCategories(categoriesWrapper));
+
+        }
+
+        // GET api/<CategoryController>/5
+        [HttpGet("{id}")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(CategoryModel), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetById(int id)
+        {
+            var categoryDto = await _categoryService.GetCategoryByIdAsync(id);
+
+            if (categoryDto == null)
+                return NotFound();
+
+            var model = CategoryModel.MapFrom(categoryDto);
+            model.Links = CreateLinksForCategory(model.Id).ToList();
+
+            return Ok(model);
+        }
+
+        [HttpGet("{id}/products")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(CategoryModel), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetProductsByCategoryId(int id, int page = 0, int limit = 50)
+        {
+            var categoryDto = await _categoryService.GetCategoryByIdAsync(id);
+
+            if (categoryDto == null)
+                return BadRequest();
+
+            var dTOs = await _productService.GetAllProductsByCategoryIdAsync(id, page, limit);
+            var models = dTOs.Select(ProductModel.MapFrom).ToList();
+
+            return Ok(models);
+        }
+
+        // POST api/<CategoryController>
+        [HttpPost]
+        [Consumes("application/json")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(CategoryModel), StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
```

Comment:
just to make it a bit more readable, please arrange ProducesResponseType attributes by ascending StatusCode